commons/protocol/src/test/resource/saved-protocol/0.2/complex/moduleid.json

```diff
@@ -0,0 +1 @@
+{"$type":"sbt.protocol.ModuleId","attributes":{"$type":"scala.collection.immutable.Map[java.lang.String,java.lang.String]","a":"b"},"name":"bar","organization":"com.foo"}
```

Comment:
what if there's a "$type" key in a Map[String,T] ? I think just omit $type for string maps, in the same way we do for lists and primitives.